src/main/java/org/apache/mesos/config/RepairConfiguration.java

```diff
@@ -0,0 +1,93 @@
+package org.apache.mesos.config;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * Represents the repair configuration in JSON.
+ */
+public class RepairConfiguration {
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+        return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        RepairConfiguration that = (RepairConfiguration) o;
+
+        if (gracePeriodMins != that.gracePeriodMins) {
+            return false;
+        }
+        if (repairDelaySecs != that.repairDelaySecs) {
+            return false;
+        }
+        return enableReplacement == that.enableReplacement;
+
+    }
+
+    @Override
+    public int hashCode() {
+        int result = gracePeriodMins;
+        result = 31 * result + repairDelaySecs;
+        result = 31 * result + (enableReplacement ? 1 : 0);
+        return result;
+    }
+
+    @JsonProperty("repairInPlaceGracePeriodMins")
+    private int gracePeriodMins;
+    @JsonProperty("minDelayBetweenRepairsSecs")
+    private int repairDelaySecs;
+    @JsonProperty("enableReplacement")
+    private boolean enableReplacement;
+
+    public  RepairConfiguration() {}
+
+    @JsonCreator
+    public RepairConfiguration(
+            @JsonProperty("repairInPlaceGracePeriodMins") int gracePeriodMins,
+            @JsonProperty("minDelayBetweenRepairsSecs") int repairDelaySecs,
+            @JsonProperty("enableReplacement") boolean enableReplacement) {
+        this.gracePeriodMins = gracePeriodMins;
+        this.repairDelaySecs = repairDelaySecs;
+        this.enableReplacement = enableReplacement;
+    }
+
+    public boolean isEnableReplacement() {
+        return enableReplacement;
+    }
+
+    @JsonProperty("enableReplacement")
+    public void setEnableReplacement(boolean enableReplacement) {
+        this.enableReplacement = enableReplacement;
+    }
+
+    public int getRepairDelaySecs() {
```

Comment:
Serdes is not a barrier--it's a style question. I would recommend using integral units, since by forcing the user to select units of a particular size, this helps to mentally anchor what appropriate choices for these values may be. 

I would strongly recommend against removing the units from the names of the properties, since those names are exposed to the YAML config file, which is totally untyped. Without the units in the name, you'd have to read the code or docs to understand the config file, rather than having it be self-documenting. Because of this, I don't think that Durations offer an advantage over `int`s.

I would, however, agree that Durations are a fine choice for programmatic APIs, where it's easy to construct whichever Duration you want. It's just that this API is interacted with via untyped YAML, so it's meant to be easier to use from there.